src/selectors.js

```diff
@@ -40,6 +40,25 @@ const getModel = (key) => {
   return { model, defaultJoins: model?.defaultJoins };
 };
 
+const flattenJoins = (joins) => {
+  let result = [];
+
+  function traverse(current) {
+    if (_.isString(current)) {
+      result.push(current);
+    } else if (Array.isArray(current) || _.isPlainObject(current)) {
+      _.forEach(current, (value, key) => {
+        if (_.isPlainObject(current)) result.push(key);
+        traverse(value);
+      });
+    }
+  }
+
+  traverse(joins);
```

Comment:
waarom definieer je traverse hier in `flattenJoins`? en doet `flattenJoins` niet dermate weinig dat het overbodig is en we alles in 1 fn kunnen stoppen?

Reply:
Omdat deze functie wordt aangroepen in findRecords. Dit heeft niet met de buildRecords te maken. De rede dat ik traverse hier aanroep is omdat hij recursief is en dus zichzelf ook weer aanroept